lib/element.js

```diff
@@ -922,8 +906,25 @@ ElementFinder.prototype.$ = function(selector) {
  *     the element is present on the page.
  */
 ElementFinder.prototype.isPresent = function() {
-  return this.parentElementArrayFinder.count().then(function(count) {
-    return !!count;
+  return this.parentElementArrayFinder.getWebElements().then(function(arr) {
+    if (arr.length == 0) {
+      return false;
+    }
+    return arr[0].isEnabled().then(function() {
+      return true; // is present, whether it is enabled or not
+    }, function(err) {
+      if (err.code == webdriver.error.ErrorCode.STALE_ELEMENT_REFERENCE) {
+        return false;
+      } else {
+        throw err;
+      }
+    });
+  }, function(err) {
+    if (err.code == webdriver.error.ErrorCode.NO_SUCH_ELEMENT) {
+      return false;
+    } else {
+      throw err;
+    }
```

Comment:
Is this any better than using https://code.google.com/p/selenium/source/browse/javascript/webdriver/webdriver.js#959?